### ISSUE TITLE
firefox: tweak UI with custom CSS

### ISSUE THREAD

**Issue by gekoke** [danth/stylix#105]
This could be implemented using [Firefox's home-manager module](https://github.com/nix-community/home-manager/blob/29519461834c08395b35f840811faf8c23e3b61c/modules/programs/firefox.nix#LL218C20-L218C20), which supports adding custom CSS.

**Comment by SomeGuyNamedMy** [comment]
can you not just set firefox to use the gtk theme?

**Comment by SomeGuyNamedMy** [comment]
it would be interesting if we could maybe set up stylix to work with the styler extension

**Comment by danth** [comment]
Firefox should use the GTK theme by default.

**Comment by Henry-Hiles** [comment]
What about firefox-gnome-theme support? Gradience has a plugin for it, and it would be a great feature!

**Comment by Henry-Hiles** [comment]
https://github.com/rafaelmardojai/firefox-gnome-theme https://github.com/GradienceTeam/Plugins/blob/main/firefox_gnome_theme.py

**Comment by Henry-Hiles** [comment]
Made a new issue for it, https://github.com/danth/stylix/issues/128!

**Comment by trueNAHO** [comment]
Potentially related: https://github.com/danth/stylix/issues/173

**Comment by W-Y-R** [comment]
The option [stylix.targets.firefox.enable](https://danth.github.io/stylix/options/hm.html#stylixtargetsfirefoxenable) from the home manager section of the wiki breaks the firefox UI, specifically the top bar.

For reference, I am on Hyprland.

**Comment by danth** [comment]
We could potentially make use of / take inspiration from these:

https://github.com/schizofox/schizofox/blob/main/modules/home-manager/firefox/userChrome.nix
https://github.com/schizofox/schizofox/blob/main/modules/home-manager/firefox/userContent.nix

**Comment by awwpotato** [comment]
this seems unnecessary since we have added firefox color support with #881

**Comment by danth** [comment]
Yes, I think the choice between Firefox GNOME Theme or Firefox Color is sufficient.
